packages/viz/__tests__/SizeBinsStyle.spec.ts

```diff
@@ -0,0 +1,166 @@
+// eslint-disable-next-line import/no-unresolved
+import { sizeBinsStyle, defaultSizeBinsOptions } from '../src/lib/style';
+
+describe('SizeBinsStyle', () => {
+  const recordExample: Record<string, any> = {
+    properties: {
+      attributeName: 12,
+      otherAttribute: 'foo'
+    }
+  };
+  const recordOutOfRangeExample: Record<string, any> = {
+    properties: {
+      attributeName: 100,
+      otherAttribute: 'foo'
+    }
+  };
+  const recordNullValueExample: Record<string, any> = {
+    properties: {
+      attributeName: null,
+      otherAttribute: 'foo'
+    }
+  };
+  describe('Style creation', () => {
+    it('should create a sizeBinsStyle instance properly', () => {
+      expect(() =>
+        sizeBinsStyle('attributeName', {
+          bins: [0, 10, 15],
+          sizes: [2, 10]
+        })
+      ).not.toThrow();
+    });
+    it('should fail when creating sizeBinsStyle instance with no bins nor sizes', () => {
+      expect(() => sizeBinsStyle('attributeName')).toThrow();
+    });
+    it('should fail when creating a sizeBinsStyle instance with invalid parameters', () => {
+      expect(
+        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+        // @ts-ignore
+        () => sizeBinsStyle('attributeName', { bins: 'invalid' })
+      ).toThrow();
+    });
+    it('should fail when creating a sizeBinsStyle instance with invalid parameters II', () => {
+      expect(() =>
+        sizeBinsStyle('attributeName', {
+          bins: [0, 10],
+          sizes: [2, 10, 15, 50]
+        })
+      ).toThrow();
```

Comment:
It could be interesting to add an expectation here about `CartoError` and some specific messages, regarding in this case to number mismatch